src/zeta/aca_oam_server.cpp

```diff
@@ -228,45 +257,54 @@ void ACA_Oam_Server::_parse_oam_flow_deletion(uint32_t in_port, oam_message *oam
   return;
 }
 
-int ACA_Oam_Server::add_direct_path(alcor::schema::NetworkType network_type, const string inst_dst_mac,
-      const string vlan_id, const string tunnel_id, const string outport_name)
+int ACA_Oam_Server::add_direct_path(oam_match match, oam_action action)
 {
   unsigned long not_care_culminative_time;
   int overall_rc = EXIT_SUCCESS;
 
-  string cmd_string = "add-flow br-tun table=0,priority=1,dl_vlan="
-            + vlan_id + "dl_dst=" + inst_dst_mac + ",actions=\"strip_vlan,load:" + tunnel_id 
-            + "->NXM_NX_TUN_ID[],output:" + outport_name + "\"";
+  string vlan_id = to_string(aca_vlan_manager::ACA_Vlan_Manager::get_instance().get_or_create_vlan_id(match.vni));
+  string outport_name = aca_get_outport_name(alcor::schema::NetworkType::VXLAN, action.node_nw_dst);
```

Comment:
you may not be able to aca_get_outport_name to get the outport once you move away from reusing create_update_neighbor_port logic mentioned above.